src/AddressRegistry.Projections.Wms/AddressWmsItem/AddressWmsItemProjections.cs

```diff
@@ -106,15 +97,25 @@ await context
 
                 if (message.Message.ParentPersistentLocalId.HasValue)
                 {
-                    await context.FindAndUpdateAddressDetail(
-                        message.Message.ParentPersistentLocalId.Value,
-                        address =>
-                        {
-                            address.LabelType = WmsAddressLabelType.BoxNumberOrHouseNumberWithBoxes;
-                        },
-                        ct,
-                        updateHouseNumberLabelsBeforeAddressUpdate: false,
-                        updateHouseNumberLabelsAfterAddressUpdate: false);
+                    var parent = await context.FindAddressDetail(message.Message.ParentPersistentLocalId.Value, ct);
+                    if (parent.Position == addressWmsItem.Position)
+                    {
+                        await context.FindAndUpdateAddressDetail(
+                            message.Message.ParentPersistentLocalId.Value,
+                            address => { address.LabelType = WmsAddressLabelType.BoxNumberOrHouseNumberWithBoxesOnSamePosition; },
+                            ct,
+                            updateHouseNumberLabelsBeforeAddressUpdate: false,
+                            updateHouseNumberLabelsAfterAddressUpdate: false);
+                    }
+                    else
+                    {
+                        await context.FindAndUpdateAddressDetail(
```

Comment:
If the position is not the same, you should not update the parent. A parent can have multiple children, so you can undo the result of a previous update.